src/ops/num.rs

```diff
@@ -0,0 +1,100 @@
+use super::word::ToWord;
+
+#[derive(Clone, Copy, Debug)]
+pub enum Num {
+  Int(i64),
+  Float(f32),
+}
+
+impl ToWord for Num {
+  fn to_word(self) -> u64 {
+    match self {
+      Self::Int(int) => int as u64,
+      Self::Float(float) => unsafe { std::mem::transmute::<_, u32>(float) as u64 },
+    }
+  }
+}
+
+pub trait Numeric: Sized {
+  const ZERO: Self;
+  const ONE: Self;
+
+  fn add(_: Self, _: Self) -> Self {
+    return Self::ZERO;
```

Comment:
you can drop the explicit `return` here

also, given how repetitive this is, I'd be inclined for it to be
```rs
#[rustfmt::skip]
trait Numeric: Sized {
  ...

  fn foo(_: Self, _: Self) -> Self { Self::ZERO }
  fn bar(_: Self, _: Self) -> Self { Self::ZERO }
  fn baz(_: Self, _: Self) -> Self { Self::ZERO }
  ...
}
```